server/source/source.controller.ts

```diff
@@ -103,6 +106,25 @@ export class SourceController {
         );
     }
 
+    @ApiTags("source")
+    @Get("/check-source")
+    @UseGuards(AbilitiesGuard)
+    @CheckAbilities(new RegularUserAbility())
```

Comment:
issue: this should be a FactChecker, regular users can't create resource anyway.